processing/src/main/java/org/apache/druid/segment/nested/GlobalDictionaryEncodedFieldColumnWriter.java

```diff
@@ -133,8 +134,15 @@ public void addValue(int row, Object val) throws IOException
       fillNull(row);
     }
     final T value = processValue(val);
-    final int globalId = lookupGlobalId(value);
-    final int localId = localDictionary.add(globalId);
+    final int localId;
+    // null is always 0
+    if (value == null) {
+      localId = localDictionary.add(0);
+    } else {
+      final int globalId = lookupGlobalId(value);
+      Preconditions.checkArgument(globalId >= 0, "Value [%s] is not present in global dictionary", value);
+      localId = localDictionary.add(globalId);
+    }
```

Comment:
Why is it not okay to do this check inside of `lookupGlobalId`?

Reply:
this is the shared function, `lookupGlobalId` has different implementations depending on the type of writer so would need to duplicate there